dandi/dandiapi.py

```diff
@@ -269,21 +287,27 @@ def dandi_authenticate(self):
                 break
 
     def get_dandiset(
-        self, dandiset_id: str, version_id: Optional[str] = None
+        self, dandiset_id: str, version_id: Optional[str] = None, lazy: bool = True
     ) -> "RemoteDandiset":
         """
         Fetches the Dandiset with the given ``dandiset_id``.  If ``version_id``
         is not specified, the `RemoteDandiset`'s version is set to the most
         recent published version if there is one, otherwise to the draft
         version.
+
+        If ``lazy`` is true, no requests are actually made until any data is
+        requested from the `RemoteDandiset`.
         """
-        d = RemoteDandiset._make(self, self.get(f"/dandisets/{dandiset_id}/"))
-        if version_id is not None and version_id != d.version_id:
-            if version_id == "draft":
-                d.version = d.draft_version
-            else:
-                return d.for_version(version_id)
-        return d
+        if lazy:
+            return RemoteDandiset(self, dandiset_id, version_id)
+        else:
+            d = RemoteDandiset._make(self, self.get(f"/dandisets/{dandiset_id}/"))
+            if version_id is not None and version_id != d.version_id:
+                if version_id == DRAFT:
+                    d.version = d.draft_version
+                else:
+                    return d.for_version(version_id)
+            return d
 
     def get_dandisets(self) -> Iterator["RemoteDandiset"]:
         for data in self.paginate("/dandisets/"):
```

Comment:
seems needing test

Reply:
Do you mean tests for `get_dandisets()`?  How should I test that?  Should the test just consume everything in the iterator and check that the elements are `RemoteDandiset`s?

Reply:
we have tests which create/populate dandisets, so it would be just a matter of adding to one of those tests that `get_dandisets` would return a list which includes the created dandiset

Reply:
Done.